src/main/java/sic/service/impl/ProductoServiceImpl.java

```diff
@@ -299,22 +299,27 @@ public List<Producto> buscarProductosParaReporte(BusquedaProductoCriteria criter
 
   @Override
   public Pageable getPageable(Integer pagina, List<String> ordenarPor, String sentido, int tamanioPagina) {
-    if (pagina == null) pagina = 0;
-    String ordenDefault = "descripcion";
-    if (ordenarPor == null || sentido == null) {
-      return PageRequest.of(pagina, tamanioPagina, Sort.by(Sort.Direction.ASC, ordenDefault));
-    } else {
-      List<Sort.Order> ordenes = new ArrayList<>();
-      switch (sentido) {
-        case "ASC":
-          ordenarPor.forEach(orden -> ordenes.add(new Sort.Order(Sort.Direction.ASC, orden)));
-          return PageRequest.of(pagina, tamanioPagina, Sort.by(ordenes));
-        case "DESC":
-          ordenarPor.forEach(orden -> ordenes.add(new Sort.Order(Sort.Direction.DESC, orden)));
-          return PageRequest.of(pagina, tamanioPagina, Sort.by(ordenes));
-        default:
-          return PageRequest.of(pagina, tamanioPagina, Sort.by(Sort.Direction.DESC, ordenDefault));
+    int numeroDePagina = pagina != null ? pagina : 0;
+    var orden = new ArrayList<Sort.Order>();
+    try {
+      if (ordenarPor != null && sentido != null) {
+        ordenarPor.forEach(nombreOrden -> {
+          var criterio = SortingProducto.fromValue(nombreOrden);
+          if (criterio == null) {
+            throw new BusinessServiceException(
+                    messageSource.getMessage("mensaje_producto_error_sorting_busqueda", null, Locale.getDefault()));
+          }
+          if (criterio == SortingProducto.FECHA_ALTA || criterio == SortingProducto.FECHA_ULTIMA_MODIFICACION) {
+            orden.add(new Sort.Order(Sort.Direction.ASC, SortingProducto.ID_PRODUCTO.getNombre()));
+          }
+          orden.add(new Sort.Order(Sort.Direction.fromString(sentido), criterio.getNombre()));
+        });
+      } else {
+        orden.add(new Sort.Order(Sort.Direction.ASC, SortingProducto.DESCRIPCION.getNombre()));
       }
+      return PageRequest.of(numeroDePagina, tamanioPagina, Sort.by(orden));
+    } catch (IllegalArgumentException illegalArgumentException) {
```

Comment:
Why this catch here?

Reply:
It was to avoid some line codes and complexity